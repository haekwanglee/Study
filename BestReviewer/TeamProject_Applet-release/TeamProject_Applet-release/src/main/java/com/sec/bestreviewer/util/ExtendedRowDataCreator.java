package com.sec.bestreviewer.util;

import com.sec.bestreviewer.database.Row;
import com.sec.bestreviewer.database.field.EmployeeSchema;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.LinkedList;
import java.util.List;

public class ExtendedRowDataCreator {
    private static final String NAME_DELIMITER = " ";
    private static final String PHONE_NUM_DELIMITER = "-";
    private static final String DATE_PATTERN = "yyyyMMdd";

    private static final int FIRST_NAME_INDEX = 0;
    private static final int LAST_NAME_INDEX = 1;

    private static final int MIDDLE_NUMBER_INDEX = 1;
    private static final int LAST_NUMBER_INDEX = 2;

    public static Row create(Row row) {
        final List<String> updatedDataList = new LinkedList<>(row.getRow());

        appendNames(row, updatedDataList);
        appendPhoneNumbers(row, updatedDataList);
        appendDates(row, updatedDataList);

        return new Row(updatedDataList);
    }

    private static void appendNames(Row row, List<String> extendedList) {
        final String name = row.getValue(EmployeeSchema.NAME.getIndex());
        final String[] splitName = name.split(NAME_DELIMITER);
        extendedList.add(splitName[FIRST_NAME_INDEX]);
        extendedList.add(splitName[LAST_NAME_INDEX]);
    }

    private static void appendPhoneNumbers(Row row, List<String> updatedDataList) {
        final String phoneNumber = row.getValue(EmployeeSchema.PHONE_NUMBER.getIndex());
        final String[] splitPhoneNumber = phoneNumber.split(PHONE_NUM_DELIMITER);
        updatedDataList.add(splitPhoneNumber[MIDDLE_NUMBER_INDEX]);
        updatedDataList.add(splitPhoneNumber[LAST_NUMBER_INDEX]);
    }

    private static void appendDates(Row row, List<String> updatedDataList) {
        final String birthDay = row.getValue(EmployeeSchema.BIRTHDAY.getIndex());
        LocalDate date = LocalDate.parse(birthDay, DateTimeFormatter.ofPattern(DATE_PATTERN));
        updatedDataList.add(String.valueOf(date.getYear()));
        updatedDataList.add(getDateString(date.getMonthValue()));
        updatedDataList.add(getDateString(date.getDayOfMonth()));
    }

    private static String getDateString(int monthOrDay) {
        String date = String.valueOf(monthOrDay);
        return date.length() == 1
                ? "0" + date
                : date;
    }
}
