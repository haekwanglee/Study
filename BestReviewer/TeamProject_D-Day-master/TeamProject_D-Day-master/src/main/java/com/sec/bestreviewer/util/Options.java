package com.sec.bestreviewer.util;

public class Options {
    public static final String LAST_NAME_OPTION = "-l";
    public static final String MIDDLE_NUMBER_OPTION = "-m";
    public static final String LAST_NUMBER_OPTION = "-l";
    public static final String YEAR_OPTION = "-y";
    public static final String MONTH_OPTION = "-m";
    public static final String DAY_OPTION = "-d";
    public static final String GRATER_OPTION = "-g";
    public static final String GRATER_EQUAL_OPTION = "-ge";
    public static final String SMALLER_OPTION = "-s";
    public static final String SMALLER_EQUAL_OPTION = "-se";
    public static final String FIRST_NAME_OPTION = "-f";
    public static final String EMPTY_OPTION = " ";
}
