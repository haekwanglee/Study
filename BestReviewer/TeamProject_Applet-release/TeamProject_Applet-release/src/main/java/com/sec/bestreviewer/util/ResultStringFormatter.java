package com.sec.bestreviewer.util;

import com.sec.bestreviewer.database.Row;
import com.sec.bestreviewer.database.field.EmployeeSchema;

import java.util.Comparator;
import java.util.List;
import java.util.stream.Collectors;

public class ResultStringFormatter {
    private static final int YEAR_1990_EMPLOYEE_NUMBER = 90_000000;
    private static final int YEAR_PREFIX_19 = 1900_000000;
    private static final int YEAR_PREFIX_20 = 2000_000000;

    public static List<String> getEmployeeListToFormattedString(List<Row> employeeList) {
        return employeeList.stream()
                .sorted(Comparator.comparing(ResultStringFormatter::getEmployeeNumber))
                .map(ResultStringFormatter::getEmployeeToFormattedString)
                .collect(Collectors.toList());
    }

    private static Integer getEmployeeNumber(Row employee) {
        int employeeNumber = Integer.parseInt(employee.getValue(EmployeeSchema.getPrimaryKeyIndex()));
        return employeeNumber >= YEAR_1990_EMPLOYEE_NUMBER ?
                YEAR_PREFIX_19 + employeeNumber : YEAR_PREFIX_20 + employeeNumber;
    }

    public static String getEmployeeToFormattedString(Row employee) {
        return employee.getValue(EmployeeSchema.EMPLOYEE_NUMBER.getIndex()) + "," +
                employee.getValue(EmployeeSchema.NAME.getIndex()) + "," +
                employee.getValue(EmployeeSchema.CAREER_LEVEL.getIndex()) + "," +
                employee.getValue(EmployeeSchema.PHONE_NUMBER.getIndex()) + "," +
                employee.getValue(EmployeeSchema.BIRTHDAY.getIndex()) + "," +
                employee.getValue(EmployeeSchema.CERTI.getIndex());

    }
}