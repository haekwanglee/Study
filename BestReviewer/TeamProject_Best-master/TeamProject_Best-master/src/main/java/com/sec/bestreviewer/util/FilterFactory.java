package com.sec.bestreviewer.util;

import com.sec.bestreviewer.store.filter.AndFilter;
import com.sec.bestreviewer.store.filter.Filter;
import com.sec.bestreviewer.store.filter.OrFilter;
import com.sec.bestreviewer.store.filter.SingleFilter;

import java.util.List;

import static com.sec.bestreviewer.util.ParamsParser.PairIndex.FIRST_PAIR;
import static com.sec.bestreviewer.util.ParamsParser.PairIndex.SECOND_PAIR;

public class FilterFactory {

    public static Filter buildFilter(List<String> params, OptionParser optionParser) {
        if (optionParser.getAndOption().isAndOn()) {
            Pair<String, String> conditionPair1 = ParamsParser.parser(params, FIRST_PAIR);
            OptionParser optionParser1 = new OptionParser(optionParser.getAndOption().getFirstOptions(), conditionPair1.fieldName);

            Pair<String, String> conditionPair2 = ParamsParser.parser(params, SECOND_PAIR);
            OptionParser optionParser2 = new OptionParser(optionParser.getAndOption().getSecondOptions(), conditionPair2.fieldName);
            return new AndFilter(new SingleFilter(conditionPair1, optionParser1), new SingleFilter(conditionPair2, optionParser2));
        }

        if (optionParser.getOrOption().isOrOn()) {
            Pair<String, String> conditionPair1 = ParamsParser.parser(params, FIRST_PAIR);
            OptionParser optionParser1 = new OptionParser(optionParser.getOrOption().getFirstOptions(), conditionPair1.fieldName);

            Pair<String, String> conditionPair2 = ParamsParser.parser(params, SECOND_PAIR);
            OptionParser optionParser2 = new OptionParser(optionParser.getOrOption().getSecondOptions(), conditionPair2.fieldName);

            return new OrFilter(new SingleFilter(conditionPair1, optionParser1), new SingleFilter(conditionPair2, optionParser2));
        }

        Pair<String, String> conditionPair = ParamsParser.parser(params, FIRST_PAIR);
        return new SingleFilter(conditionPair, optionParser);
    }
}
