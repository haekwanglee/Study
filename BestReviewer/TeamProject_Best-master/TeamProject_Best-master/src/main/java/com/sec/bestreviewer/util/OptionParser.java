package com.sec.bestreviewer.util;

import com.sec.bestreviewer.option.*;

import java.util.Arrays;
import java.util.List;

public class OptionParser {

    public final static List<String> SUPPORTED_OPTIONS = Arrays.asList("-p", "-f", "-l", "-m", "-y", "-d", "-g", "-ge", "-s", "-se", "-a", "-o");
    private final PrintOption printOption;
    private final NameOption nameOption;
    private final BirthDayOption birthDayOption;
    private final PhoneOption phoneOption;
    private final EqualityOption equalityOption;
    private final AndOption andOption;
    private final OrOption orOption;

    public OptionParser(List<String> options, String fieldName) {
        printOption = new PrintOption(options);
        nameOption = new NameOption(options, fieldName);
        birthDayOption = new BirthDayOption(options, fieldName);
        phoneOption = new PhoneOption(options, fieldName);
        equalityOption = new EqualityOption(options);
        andOption = new AndOption(options);
        orOption = new OrOption(options);
    }

    public PrintOption getPrintOption() {
        return printOption;
    }

    public NameOption getNameOption() {
        return nameOption;
    }

    public BirthDayOption getBirthDayOption() {
        return birthDayOption;
    }

    public PhoneOption getPhoneOption() {
        return phoneOption;
    }

    public EqualityOption getEqualityOption() {
        return equalityOption;
    }

    public AndOption getAndOption() {
        return andOption;
    }

    public OrOption getOrOption() {
        return orOption;
    }
}
