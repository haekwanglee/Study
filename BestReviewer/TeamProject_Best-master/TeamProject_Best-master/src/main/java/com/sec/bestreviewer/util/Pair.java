package com.sec.bestreviewer.util;

import java.util.Objects;

public class Pair<F, S> {
    public final F fieldName;
    public final S value;

    private Pair(F fieldName, S value) {
        this.fieldName = fieldName;
        this.value = value;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Pair)) {
            return false;
        }
        Pair<?, ?> p = (Pair<?, ?>) o;
        return Objects.equals(p.fieldName, fieldName) && Objects.equals(p.value, value);
    }

    @Override
    public int hashCode() {
        return (fieldName == null ? 0 : fieldName.hashCode()) ^ (value == null ? 0 : value.hashCode());
    }

    @Override
    public String toString() {
        return "Pair{" + String.valueOf(fieldName) + " " + String.valueOf(value) + "}";
    }

    public static <A, B> Pair <A, B> create(A a, B b) {
        return new Pair<A, B>(a, b);
    }
}