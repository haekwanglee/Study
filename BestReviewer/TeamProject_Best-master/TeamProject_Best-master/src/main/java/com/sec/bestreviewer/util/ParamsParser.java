package com.sec.bestreviewer.util;

import com.sec.bestreviewer.store.EmployeeStore;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class ParamsParser {

    public enum PairIndex {
        FIRST_PAIR(0), SECOND_PAIR(2), THIRD_PAIR(4);

        private final int index;
        PairIndex(int index) {
            this.index = index;
        }
    }

    private static final Map<String, String> fieldMap = new HashMap<>();
    private static final String EMPLOYEE_NUMBER = "employeeNum";
    private static final String NAME = "name";
    private static final String CAREER_LEVEL = "cl";
    private static final String PHONE_NUMBER = "phoneNum";
    private static final String BIRTHDAY = "birthday";
    private static final String CERTI = "certi";

    // mapping table between commands name and DB columns name
    static {
        fieldMap.put(EMPLOYEE_NUMBER, EmployeeStore.FIELD_EMPLOYEE_NUMBER);
        fieldMap.put(NAME, EmployeeStore.FIELD_NAME);
        fieldMap.put(CAREER_LEVEL, EmployeeStore.FIELD_CAREER_LEVEL);
        fieldMap.put(PHONE_NUMBER, EmployeeStore.FIELD_PHONE_NUMBER);
        fieldMap.put(BIRTHDAY, EmployeeStore.FIELD_BIRTH_DAY);
        fieldMap.put(CERTI, EmployeeStore.FIELD_CERTI);
    }

    public static Pair<String, String> parser(List<String> params, PairIndex pairIndex) {
        return getMapFromParams(params, pairIndex.index);
    }

    private static Pair<String, String> getMapFromParams(List<String> params, int index) {
        try {
            final String fieldName = fieldMap.get(params.get(index));
            if (fieldName == null) {
                throw new IllegalArgumentException("Wrong field: " + params.get(index));
            }
        } catch (ArrayIndexOutOfBoundsException ex) {
            throw new IllegalArgumentException("Wrong parameters");
        }
        return Pair.create(fieldMap.get(params.get(index)), params.get(index + 1));
    }
}
