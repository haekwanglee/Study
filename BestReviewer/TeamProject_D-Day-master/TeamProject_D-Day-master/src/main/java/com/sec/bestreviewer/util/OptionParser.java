package com.sec.bestreviewer.util;

import java.util.List;

public class OptionParser {

    private static final int NUMBER_OF_OPTION_SIZE_FOR_FIRST_COMMAND = 3;   //first command consists of '-p', '-f/l/y/m/d/ln/mn', '-g/ge/s/se'
    private static final int NUMBER_OF_OPTION_SIZE_FOR_SECOND_COMMAND = 2;  //second command consists of '-f/l/y/m/d/ln/mn', '-g/ge/s/se'

    private boolean isPrintOn = false;

    private String filterOption = Options.EMPTY_OPTION;
    private String compareOption = Options.EMPTY_OPTION;

    public OptionParser(List<String> options) {
        if (!options.isEmpty()) {
            initializeByNumberOfOptions(options);
        }
    }

    private void initializeByNumberOfOptions(List<String> options) {
        if (options.size() == NUMBER_OF_OPTION_SIZE_FOR_FIRST_COMMAND)
            initFirstCommandOptionParser(options);
        else if (options.size() == NUMBER_OF_OPTION_SIZE_FOR_SECOND_COMMAND)
            initSecondCommandOptionParser(options);
        else
            throw new IllegalArgumentException("Invalid number of options.");
    }

    private void initFirstCommandOptionParser(List<String> options) {
        setPrintOption(options.get(0));
        setFilterOption(options.get(1));
        setCompareOption(options.get(2));
    }

    private void initSecondCommandOptionParser(List<String> options) {
        setFilterOption(options.get(0));
        setCompareOption(options.get(1));
    }

    private void setPrintOption(String commandPrintArgument) {
        if (isNotEmptyOption(commandPrintArgument))
            isPrintOn = true;
    }

    private void setFilterOption(String commandFirstArgument) {
        if (isNotEmptyOption(commandFirstArgument)) {
            filterOption = commandFirstArgument;
        }
    }

    private void setCompareOption(String commandSecondArgument) {
        if (isNotEmptyOption(commandSecondArgument)) {
            compareOption = commandSecondArgument;
        }
    }

    private boolean isNotEmptyOption(String optionValue) {
        return !optionValue.equals(Options.EMPTY_OPTION);
    }

    public boolean isPrintOn() {
        return isPrintOn;
    }

    public String getFilterOption() {
        return filterOption;
    }

    public String getCompareOption() {
        return compareOption;
    }
}
