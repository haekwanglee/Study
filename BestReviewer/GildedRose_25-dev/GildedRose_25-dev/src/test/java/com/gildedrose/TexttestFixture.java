package com.gildedrose;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.List;

public class TexttestFixture {

    @Test
    @DisplayName("Golden Master Test")
    public void goldenMasterTest() {
        System.out.println("OMGHAI!");

        Item[] items = new Item[] {
                Item.getItem("+5 Dexterity Vest", 10, 20), //
                Item.getItem("Aged Brie", 2, 0), //
                Item.getItem("Elixir of the Mongoose", 5, 7), //
                Item.getItem("Sulfuras, Hand of Ragnaros", 0, 80), //
                Item.getItem("Sulfuras, Hand of Ragnaros", -1, 80),
                Item.getItem("Backstage passes to a TAFKAL80ETC concert", 15, 20),
                Item.getItem("Backstage passes to a TAFKAL80ETC concert", 10, 49),
                Item.getItem("Backstage passes to a TAFKAL80ETC concert", 5, 49),
                // this conjured item does not work properly yet
                Item.getItem("Conjured Mana Cake", 3, 6) };

        GildedRose app = new GildedRose(items);

        int days = 2;
        String TAG = "goldenMasterTest";
        List<String> testResults = new ArrayList<>();
        for (int i = 0; i < days; i++) {
            testResults.add("-------- day " + i + " --------");
            testResults.add("name, sellIn, quality");
            for (Item item : items) {
                testResults.add(item.toString());
            }
            testResults.add("");
            app.updateQuality();
        }
        Approvals.verifyAll(TAG, testResults.toArray());
    }

}
