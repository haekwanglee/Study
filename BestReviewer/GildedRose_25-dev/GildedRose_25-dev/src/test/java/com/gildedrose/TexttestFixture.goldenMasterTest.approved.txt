goldenMasterTest[00] = -------- day 0 --------
goldenMasterTest[01] = name, sellIn, quality
goldenMasterTest[02] = +5 Dexterity Vest, 10, 20
goldenMasterTest[03] = Aged Brie, 2, 0
goldenMasterTest[04] = Elixir of the Mongoose, 5, 7
goldenMasterTest[05] = Sulfuras, Hand of Ragnaros, 0, 80
goldenMasterTest[06] = Sulfuras, Hand of Ragnaros, -1, 80
goldenMasterTest[07] = Backstage passes to a TAFKAL80ETC concert, 15, 20
goldenMasterTest[08] = Backstage passes to a TAFKAL80ETC concert, 10, 49
goldenMasterTest[09] = Backstage passes to a TAFKAL80ETC concert, 5, 49
goldenMasterTest[10] = Conjured Mana Cake, 3, 6
goldenMasterTest[11] = 
goldenMasterTest[12] = -------- day 1 --------
goldenMasterTest[13] = name, sellIn, quality
goldenMasterTest[14] = +5 Dexterity Vest, 9, 19
goldenMasterTest[15] = Aged Brie, 1, 1
goldenMasterTest[16] = Elixir of the Mongoose, 4, 6
goldenMasterTest[17] = Sulfuras, Hand of Ragnaros, 0, 80
goldenMasterTest[18] = Sulfuras, Hand of Ragnaros, -1, 80
goldenMasterTest[19] = Backstage passes to a TAFKAL80ETC concert, 14, 21
goldenMasterTest[20] = Backstage passes to a TAFKAL80ETC concert, 9, 50
goldenMasterTest[21] = Backstage passes to a TAFKAL80ETC concert, 4, 50
goldenMasterTest[22] = Conjured Mana Cake, 2, 5
goldenMasterTest[23] = 
